ci_framework/roles/env_op_images/tasks/main.yml

```diff
@@ -0,0 +1,91 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directory is present
+  ansible.builtin.file:
+    path: "{{ cifmw_env_op_images_dir }}/{{ item }}"
+    state: directory
+  loop:
+    - artifacts
+    - logs
+
+- name: Check if OpenStackControlPlane is setup
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell: |
+    oc get OpenStackControlPlane -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} --no-headers=true | awk '{print $2}'
+  register: openstack_installed
+
+- name: Get openstack operator images
+  when:
+    - openstack_installed.stdout == "True"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ignore_errors: true
+  block:
+    - name: Create a file to store the images
+      ansible.builtin.copy:
+        dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file  }}"
+        content: "cifmw_install_yamls_vars:"
+
+    - name: Get openstack operator-index image
```

Comment:
Why not making a single call instead of 4x2? One single call to fetch those pods as yaml will be enough, with the retrieved data we can post process it here using plain Ansible. That way we can reduce the number of calls to kubeapi and make the whole snippet more Ansible friendly instead of having a tone of bash shells. If you want to fetch the data in a single call you can try the k8s module, that fetches the content directly as an Ansible variable, or use `oc` and parse with a `from_yaml`/`from_json` afterwards.

Thank way you will end up with a proper formated file with all the needed info in a single api call. With a format like this one:
```yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: Pod
  metadata:
  ... pod meta
  spec:
   ... pod spec
  status:
    conditions:
     ... pod conditions
    containerStatuses:
    - containerID: cri-o://83e932c9a1672cd14c4f05792ffb4d1c96f264d4b0bb55121f601cc01a78cde7
      image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:7370b4b4acca491815ac6e617fa04aff824c62ad1e94e19d55505ba510539798
      imageID: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:7370b4b4acca491815ac6e617fa04aff824c62ad1e94e19d55505ba510539798
      lastState: {}
      name: extract
      ready: false
      restartCount: 0
      started: false
      state:
        terminated:
          containerID: cri-o://83e932c9a1672cd14c4f05792ffb4d1c96f264d4b0bb55121f601cc01a78cde7
          exitCode: 0
          finishedAt: "2023-08-24T09:02:29Z"
          reason: Completed
          startedAt: "2023-08-24T09:02:27Z"
    hostIP: 192.168.126.11
    initContainerStatuses:
    ... init container status
    phase: Succeeded
    podIP: 10.217.0.80
    podIPs:
    - ip: 10.217.0.80
    qosClass: Burstable
    startTime: "2023-08-24T09:01:53Z"
- apiVersion: v1
  kind: Pod
  metadata:
    annotations:`
... other pod data
```
Using plain Ansible filters like select/selectattr with equalto and match will be enough to correctly point to the needed field for every Pod.